quesma/frontend_connectors/elasticsearch_query.go

```diff
@@ -0,0 +1,158 @@
+		metadata := quesma_api.MakeNewMetadata()
+		metadata[IndexPattern] = getIndexPatternFromRequestURI(req.OriginalRequest, ResolveIndexPath)
+		metadata[PathPattern] = ResolveIndexPath
+		return &quesma_api.Result{Meta: metadata, GenericResult: req.OriginalRequest}, nil
+	})
+	router.Register(ClusterHealthPath, quesma_api.IsHTTPMethod("GET"), func(ctx context.Context, req *quesma_api.Request, writer http.ResponseWriter) (*quesma_api.Result, error) {
+		metadata := quesma_api.MakeNewMetadata()
+		metadata[PathPattern] = ClusterHealthPath
+		return &quesma_api.Result{Meta: metadata, GenericResult: req.OriginalRequest}, nil
+	})
+	router.Register(IndexRefreshPath, quesma_api.IsHTTPMethod("POST"), func(ctx context.Context, req *quesma_api.Request, writer http.ResponseWriter) (*quesma_api.Result, error) {
+		metadata := quesma_api.MakeNewMetadata()
+		metadata[IndexPattern] = getIndexPatternFromRequestURI(req.OriginalRequest, IndexRefreshPath)
+		metadata[PathPattern] = IndexRefreshPath
+		return &quesma_api.Result{Meta: metadata, GenericResult: req.OriginalRequest}, nil
+	})
+	router.Register(IndexMappingPath, quesma_api.IsHTTPMethod("GET"), func(ctx context.Context, req *quesma_api.Request, writer http.ResponseWriter) (*quesma_api.Result, error) {
+		metadata := quesma_api.MakeNewMetadata()
+		metadata[IndexPattern] = getIndexPatternFromRequestURI(req.OriginalRequest, IndexRefreshPath)
+		metadata[PathPattern] = IndexMappingPath
+		return &quesma_api.Result{Meta: metadata, GenericResult: req.OriginalRequest}, nil
+	})
+	router.Register(TermsEnumPath, quesma_api.IsHTTPMethod("POST"), func(ctx context.Context, req *quesma_api.Request, writer http.ResponseWriter) (*quesma_api.Result, error) {
+		metadata := quesma_api.MakeNewMetadata()
+		metadata[IndexPattern] = getIndexPatternFromRequestURI(req.OriginalRequest, TermsEnumPath)
+		metadata[PathPattern] = TermsEnumPath
+		return &quesma_api.Result{Meta: metadata, GenericResult: req.OriginalRequest}, nil
+	})
+	router.Register(IndexCountPath, quesma_api.IsHTTPMethod("GET"), func(ctx context.Context, req *quesma_api.Request, writer http.ResponseWriter) (*quesma_api.Result, error) {
+		metadata := quesma_api.MakeNewMetadata()
+		metadata[IndexPattern] = getIndexPatternFromRequestURI(req.OriginalRequest, IndexCountPath)
+		metadata[PathPattern] = IndexCountPath
+		return &quesma_api.Result{Meta: metadata, GenericResult: req.OriginalRequest}, nil
+	})
+
+	router.Register("*", quesma_api.Always(), func(ctx context.Context, req *quesma_api.Request, writer http.ResponseWriter) (*quesma_api.Result, error) {
+		metadata := quesma_api.MakeNewMetadata()
+		metadata[Bypass] = true
+		return &quesma_api.Result{Meta: metadata, GenericResult: req.OriginalRequest}, nil
+	})
+
+	fc.AddRouter(router)
+	return fc
+}
+
+func getIndexPatternFromRequestURI(request *http.Request, indexPath string) string {
```

Comment:
These two functions are almost identical. They differ only in return statement, so we can merge two first lines into common function

Reply:
https://github.com/QuesmaOrg/quesma/pull/1119/commits/9c950fce782956b47b1433a954895b11504a00ce